### PR TITLE
Alerting: All notification channels should send even if one fails to send

### LINKED ISSUE #19768: Alerting: All notification channels should send even if one fails to send
**What happened:**
A Grafana alert was configured with the following notifications in the sequence below:

Slack
Alerta
Email

From the logs we can see the Slack notification sent, then we can see the Alerta notification failed (a "500 Server Error" from the webhook). No email notification was ever attempted.

Error received
```
t=2019-10-09T04:13:05+0100 lvl=dbug msg="Webhook failed" logger=notifications statuscode="500 INTERNAL SERVER ERROR" body="{\"code\":500,\"errors\":null,\"messa
ge\":\"Logstash TCP connection error: [Errno 111] Connection refused\",\"requestId\":\"Root=1-5d9d5041-b8ca91f6917a8d05909bb2bc\",\"status\":\"error\"}\n"
```
After receiving the error no attempt was made to send the email alert.

**What you expected to happen:**
After the failed Alerta notification I expected the email alert to fire. 

**Environment:**
- Grafana version: v6.4.1 (4ba8388)
- Data source type & version:
- OS Grafana is installed on: Ubuntu 16.04
- User OS & Browser: macOS Mojave 10.14.6

### PR DESCRIPTION
**Pull request by jan25**
Signed-off-by: Abhilash Gnan <abhilashgnan@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #19768

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/alerting/notifier.go [hunk 1]
```diff
@@ -34,14 +34,15 @@ type notificationService struct {
 	renderService rendering.Service
 }
 
-func (n *notificationService) SendIfNeeded(context *EvalContext) error {
+func (n *notificationService) SendIfNeeded(context *EvalContext) {
```
Comment (marefr) [inline pkg/services/alerting/notifier.go]: Please add back the return of the error. We're currently doing some other error handling refactorings which handles the error returned from this method.

File: pkg/services/alerting/notifier.go [hunk 2]
```diff
@@ -34,14 +34,15 @@ type notificationService struct {
...
 	notifierStates, err := n.getNeededNotifiers(context.Rule.OrgID, context.Rule.Notifications, context)
 	if err != nil {
-		return err
+		n.log.Error("Failed to get alert notifiers", "error", err)
```
Comment (marefr) [inline pkg/services/alerting/notifier.go]: Good that you added error logging here since there's currently no handling of the error returned :+1:

**Review by marefr — CHANGES_REQUESTED** [review]
Thanks. Looks great, but have a few minor requests for changes.

**Comment by marefr** [comment]
We'll probably release v6.4.3 tomorrow and if I haven't heard back from you we'll probably push the requested changes to this branch and merge the PR since we want to get it out in v6.4.3,

**Comment by jan25** [comment]
@marefr thanks for the review! Could you restart the CI? looks like circleCI server issues.

**Review by marefr — APPROVED** [review]
Thanks. LGTM

**Comment by marefr** [comment]
Thank you for contributing to Grafana!

**Comment by marefr** [comment]
@jan25 Totally optional, but may I ask you if you're participating in Hacktoberfest and whether you've find this issue through the hacktoberfest label? Trying to find out if new contributors are finding Grafana due to Hacktoberfest.

**Comment by jan25** [comment]
Hey @marefr ! Yes i'm participating in Hacktoberfest for first time. I'm not new to Grafana project though, I was browsing through the Grafana issues list in this repo when i came across this issue. Is Grafana linked somewhere on Hacktoberfest website/twitter? that'd be super helpful for new contributors

**Comment by marefr** [comment]
@jan25 Thanks. Nice to hear. I thought Hacktoberfest site linked to GitHub global search for example filtering on opening issues labeled Hacktoberfest, see https://github.com/search?q=label%3AHacktoberfest&state=open&type=Issues. Besides this [Hacktoberfest site](https://hacktoberfest.digitalocean.com/details) gives some examples of external sites where you can find issues to work on (more beginner friendly nature). 

Again, thanks for contributing to Grafana.

**Pull request merged**